### PR TITLE
feature/521: theme private sharing

### PR DESCRIPTION
**Pull request by csanyiarpad**
- Indication whether the node is public or private.
- The switch is currently linked (with js) to the edit page. It needs
  rewrite to ajax.
- Themed sharing links, texts, and buttons.
- There is a related commit with css and fonticon changes in the main
  repository.

### CODE REVIEW

File: walkthrough_sharelink/walkthrough_sharelink.module [hunk 1]
```diff
@@ -238,30 +239,63 @@ function walkthrough_sharelink_node_view_alter(&$build) {
...
+  $build['sharelink'] = array(
+    '#type' => 'container',
+    '#attributes' => array(
+      'class' => array('walkthrough-sharelink-container'),
+    ),
+    '#weight' => -10,
+  );
+  $build['sharelink']['public_private_switch'] = array(
+    '#type' => 'markup',
+    '#markup' => '<span class="icon-unlock"></span>
```
Comment (tamasd) [inline walkthrough_sharelink/walkthrough_sharelink.module]: This should be a separate theme function and probably a template.

File: walkthrough_sharelink/walkthrough_sharelink.module [hunk 2]
```diff
@@ -238,30 +239,63 @@ function walkthrough_sharelink_node_view_alter(&$build) {
...
   if ($access_key) {
-    $build['sharelink'] = _walkthrough_sharelink_get_link_render_array($build['#node'], $access_key) + $prefix_suffix;
+    $build['sharelink']['sharelink'] = _walkthrough_sharelink_get_link_render_array($build['#node'], $access_key) + $prefix_suffix_weight;
   }
   else {
-    $build['sharelink'] = array(
+    $build['sharelink']['sharelink'] = array(
       '#theme' => 'link',
-      '#text' => t('Share this walkthrough with a link'),
+      '#text' => t(' Share this walkthrough with a link'),
```
Comment (Coornail) [inline walkthrough_sharelink/walkthrough_sharelink.module]: That's probably not a good idea

File: walkthrough_sharelink/walkthrough_sharelink.module [hunk 3]
```diff
@@ -272,7 +306,13 @@ function walkthrough_sharelink_node_view_alter(&$build) {
           array('system', 'drupal.ajax'),
         ),
       ),
-    ) + $prefix_suffix;
+    ) + $prefix_suffix_weight;
+
+    $build['sharelink']['note'] = array(
+      '#type' => 'markup',
+      '#markup' => '<p class="sharing-notice private"> Or you can make it public, to share with everyone free of charge.</p>',
```
Comment (Coornail) [inline walkthrough_sharelink/walkthrough_sharelink.module]: Missing `t()`

Comment (tamasd) [inline walkthrough_sharelink/walkthrough_sharelink.module]: Theme function or template is advised here too

**Pull request merged**